packages/vscode-dependencies-validation/test/vscodeMocks.ts

```diff
@@ -0,0 +1,189 @@
+import { NPM_DEPENDENCY_ISSUES_CODE } from "../src/constants";
+import type {
+  Diagnostic,
+  DiagnosticCollection,
+  Uri,
+  CancellationToken,
+  CodeActionContext,
+  Event,
+  Position,
+  Range,
+  TextDocument,
+  TextLine,
+  CodeActionKind,
+} from "vscode";
+import { VscodeOutputChannel } from "../src/vscodeTypes";
+
+export const outputChannel: VscodeOutputChannel = {
+  append(value: string) {},
+  appendLine(value: string) {},
+  // @ts-expect-error -- https://stackoverflow.com/questions/68799234/typescript-pick-only-specific-method-from-overload-to-be-passed-to-parameterst
+  show(value?: boolean) {},
+};
+
+export const diagnosticCollection: DiagnosticCollection = {
+  name: "",
+  // @ts-expect-error -- https://stackoverflow.com/questions/68799234/typescript-pick-only-specific-method-from-overload-to-be-passed-to-parameterst
+  set: function (
+    uri: Uri,
+    diagnostics: readonly Diagnostic[] | undefined
+  ): void {
+    //throw new Error("Function not implemented.");
+  },
+  delete: function (uri: Uri): void {
+    throw new Error("Function not implemented.");
+  },
+  clear: function (): void {
+    throw new Error("Function not implemented.");
+  },
+  forEach: function (
+    callback: (
+      uri: Uri,
+      diagnostics: readonly Diagnostic[],
+      collection: DiagnosticCollection
+    ) => any,
+    thisArg?: any
+  ): void {
+    throw new Error("Function not implemented.");
+  },
+  get: function (uri: Uri): readonly Diagnostic[] | undefined {
+    throw new Error("Function not implemented.");
+  },
+  has: function (uri: Uri): boolean {
+    throw new Error("Function not implemented.");
+  },
+  dispose: function (): void {
+    throw new Error("Function not implemented.");
+  },
+};
+
+export const textDocument: TextDocument = {
```

Comment:
does this contain what we need for TextDocument?
https://www.npmjs.com/package/vscode-languageserver-textdocument